### PR TITLE
Use cloud-prepare for AWS provisioning

### PR DESCRIPTION
**Pull request by skitt**
This is a first step towards delegating all cloud provisioning to
cloud-prepare. For now, we only set up the cloud-prepare client; all
the actual AWS configuration still relies on the addon-specific code.

The Submariner dependencies need to be updated to the current devel
versions (with the associated transitive dependencies) to work with
the K8s 1.19+ APIs.

Signed-off-by: Stephen Kitt <skitt@redhat.com>

### CODE REVIEW

File: pkg/cloud/aws/aws.go [hunk 1]
```diff
@@ -63,9 +67,11 @@ type awsProvider struct {
 	clusterName   string
 	instanceType  string
 	gateways      int
+	cloudPrepare  cpapi.Cloud
 }
 
 func NewAWSProvider(
+	restConfig *rest.Config,
```
Comment (tpantelis) [inline pkg/cloud/aws/aws.go:74]: Why is this passed in? It looks unused. `rest.Config` is problematic for unit tests in terms of mocking so would be ideal if we can avoid passing it in code we want to unit test.

**Review by tpantelis — CHANGES_REQUESTED** [review]

File: pkg/cloud/aws/aws.go (hunk 1 above)
Comment (skitt) [reply pkg/cloud/aws/aws.go:74]: Ah yes, thanks, it was needed with a previous version of cloud-prepare (with the MachineSetDeployer) but is no longer necessary.

**Review by tpantelis — APPROVED** [review]

**Pull request merged**